src/clj_3df/compiler.cljc

```diff
@@ -550,7 +563,7 @@
         aggregation (->> (:find ir)
                          extract-aggregations
                          (reduce (fn [unified {:keys [aggregation-fn vars]}]
-                                   (->Aggregation aggregation-fn vars unified)) unified))
+                                   (->Aggregation aggregation-fn vars (extract-key-symbols (:find ir)) unified (extract-find-symbols (:find ir)))) unified))
```

Comment:
Looks like `extract-key-symbols` could be done once, outside of the reduction?

Reply:
True.
Why not ditch the whole reduce of extract-aggregation and not allow multiple aggregations in the :find clause for now, until we have a better backend representation.
The way we have it implemented leads to nested `:Aggregation` calls which don't make sense in the backend when the dataflow is constructed.
We feed the output of one aggregation into another which leads in the best case to error and in the worst to panics.
Multiple aggregations would need  a duplication and subsequent joining of the underlying dataflows create by the bindings.
Or am I missing something?

Reply:
Fine with me. We can look at something like `(->Aggregation [fn vars ...] [fn vars ...] ...+)`. Should also allow for the most efficient implementations (e.g. re-use the same grouping and compute multiple aggregates per group).

But removing the reduce for now and throwing on more than one aggregation makes sense for now.